### PR TITLE
RnD Optimization

### PR DESCRIPTION
**Pull request by AnturK**
Should be good. The actual system changes can be done separately.

### CODE REVIEW

File: code/modules/research/rdconsole.dm [hunk 1]
```diff
@@ -570,7 +548,8 @@ proc/CallMaterialName(ID)
 			compare = IMPRINTER
 			screen = 4.17
```
Comment (neersighted) [inline code/modules/research/rdconsole.dm]: aaaaaa what is this magic number

**Comment by neersighted** [comment]
Looks good and cleans up a lot of shit code... It still shit but now its less shit so :+1:

File: code/modules/research/rdconsole.dm [hunk 2]
```diff
@@ -88,16 +78,12 @@ proc/CallMaterialName(ID)
...
 		for(var/R in subtypesof(/datum/reagent))
-			temp_reagent = null
-			temp_reagent = new R()
-			if(temp_reagent.id == ID)
-				return_name = temp_reagent.name
-				qdel(temp_reagent)
-				temp_reagent = null
-				break
-	return return_name
+			var/datum/reagent/rr = R
```
Comment (RemieRichards) [inline code/modules/research/rdconsole.dm]: this is funny cos I popularised this loop format here.
and it's my initials :P

File: code/modules/research/rdconsole.dm (hunk 1 above)
Comment (Menshin) [inline code/modules/research/rdconsole.dm]: Iirc this was used in legacy code to simulate tabbing or sub-category screen (pretty sure i shamelessly added to the mess when adding category grouping and search functionality to the lathes).
A strong candidate to tgui converting i guess.

**Comment by Fox-McCloud** [comment]
be careful about using TRUE and FALSE; we've personally had problems with BYOND arbitrarily deciding that TRUE wasn't, in fact, 1 anymore.

**Comment by Razharas** [comment]
>AddDesign2Known finally being fixed

Its like new feature freeze is upon us

**Pull request merged**